### ISSUE TITLE
[MNG-4028] Maven 2 still absolutely does not work

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#5966]
**[Richard Katz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rich-katz)** opened **[MNG-4028](https://issues.apache.org/jira/browse/MNG-4028?redirect=false)** and commented

I originally reported this on Maven 2.0.9 and it was marked as closed.

But the problem still exists in Maven 2.1.0.  I have tried the following
1. Delete the .m2 directory and all its contents.
2. Use the -U option
3. Use the -e option.
4. Run the create Archtype

mvn -e archetype:create -DgroupId=com.mycompany.app  -DartifactId=my-app
(and mvn -e -U archetype:create -DgroupId=com.mycompany.app  -DartifactId=my-app)

It simply does not work.

It fails with the same error as I always get in Maven 2.0.9 :

org.apache.maven.lifecycle.LifecycleExecutionException: The plugin 'org.apache.maven.plugins:maven-archetype-plugin' does not exist or no valid version could be found

Here is the entire trace:
C:\Documents and Settings\Richard Katz>mvn -e -U archetype:create -DgroupId=com.mycompany.app  -DartifactId=my-app
+ Error stacktraces are turned on.
  [INFO] Scanning for projects...
  [INFO] Searching repository for plugin with prefix: 'archetype'.
  [INFO] org.apache.maven.plugins: checking for updates from central
  [WARNING] repository metadata for: 'org.apache.maven.plugins' could not be retrieved from repository: central due to an
  error: Error transferring file
  [INFO] Repository 'central' will be blacklisted
  [INFO] ------------------------------------------------------------------------
  [ERROR] BUILD ERROR
  [INFO] ------------------------------------------------------------------------
  [INFO] The plugin 'org.apache.maven.plugins:maven-archetype-plugin' does not exist or no valid version could be found
  [INFO] ------------------------------------------------------------------------
  [INFO] Trace
  org.apache.maven.lifecycle.LifecycleExecutionException: The plugin 'org.apache.maven.plugins:maven-archetype-plugin' doe
  s not exist or no valid version could be found
  at org.apache.maven.lifecycle.DefaultLifecycleExecutor.verifyPlugin(DefaultLifecycleExecutor.java:1303)
  at org.apache.maven.lifecycle.DefaultLifecycleExecutor.getMojoDescriptor(DefaultLifecycleExecutor.java:1542)
  at org.apache.maven.lifecycle.DefaultLifecycleExecutor.segmentTaskListByAggregationNeeds(DefaultLifecycleExecuto
  r.java:405)
  at org.apache.maven.lifecycle.DefaultLifecycleExecutor.execute(DefaultLifecycleExecutor.java:137)
  at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:336)
  at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:129)
  at org.apache.maven.cli.MavenCli.main(MavenCli.java:287)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
  at java.lang.reflect.Method.invoke(Method.java:597)
  at org.codehaus.classworlds.Launcher.launchEnhanced(Launcher.java:315)
  at org.codehaus.classworlds.Launcher.launch(Launcher.java:255)
  at org.codehaus.classworlds.Launcher.mainWithExitCode(Launcher.java:430)
  at org.codehaus.classworlds.Launcher.main(Launcher.java:375)
  Caused by: org.apache.maven.plugin.version.PluginVersionNotFoundException: The plugin 'org.apache.maven.plugins:maven-ar
  chetype-plugin' does not exist or no valid version could be found
  at org.apache.maven.plugin.version.DefaultPluginVersionManager.resolvePluginVersion(DefaultPluginVersionManager.
  java:229)
  at org.apache.maven.plugin.version.DefaultPluginVersionManager.resolvePluginVersion(DefaultPluginVersionManager.
  java:91)
  at org.apache.maven.plugin.DefaultPluginManager.verifyPlugin(DefaultPluginManager.java:171)
  at org.apache.maven.lifecycle.DefaultLifecycleExecutor.verifyPlugin(DefaultLifecycleExecutor.java:1274)
  ... 14 more
  [INFO] ------------------------------------------------------------------------
  [INFO] Total time: 2 seconds
  [INFO] Finished at: Tue Feb 10 21:40:41 PST 2009
  [INFO] Final Memory: 1M/4M
  [INFO] ------------------------------------------------------------------------

C:\Documents and Settings\Richard Katz>

Questions:

1. Is this enough information?  If not, what other command or commands do you want me to run?
2. Is there amy way around this problem?  Is there a fix in 2.1 (or 2.1 M2) that hasn't been applied yet in 2.1 M1?

---

**Affects:** 2.1.0-M1

**Attachments:**
- [screenshot-1.jpg](https://issues.apache.org/jira/secure/attachment/12712831/screenshot-1.jpg) (_119.44 kB_)
- [settings.xml](https://issues.apache.org/jira/secure/attachment/12712874/settings.xml) (_279 bytes_)

**Comment by jira-importer** [comment]
**[Brett Porter](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brettporter)** commented

notice this line:

WARNING] repository metadata for: 'org.apache.maven.plugins' could not be retrieved from repository: central due to an
error: Error transferring file

Your machine is not able to connect to the remote repository (as described here: http://docs.codehaus.org/display/MAVENUSER/FAQs-1#FAQs-1-novalidversion)

You can get further connection diagnostics surrounding that line using -X, but the problem is certainly an inability to retrieve remote files.

**Comment by jira-importer** [comment]
**[Richard Katz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rich-katz)** commented

Execution of mvn -v

---

C:\Documents and Settings\Richard Katz>mvn -v
Maven version: 2.1.0-M1
Java version: 1.6.0_01
Default locale: en_US, platform encoding: Cp1252
OS name: "windows xp" version: "5.1" arch: "x86" family: "windows"

---

I have also attempted the command after inserting (and after removing) the following file in .m2 as settings.xml

---

\<settings>
\<mirrors>
\<mirror>
\<id>ibiblio.org\</id>
\<name>ibiblio Mirror of http://repo1.maven.org/maven2/\</name>
\<url>http://mirrors.ibiblio.org/pub/mirrors/maven2\</url>
\<mirrorOf>central\</mirrorOf>
\</mirror>
\</mirrors>
\</settings>

---

All set variables:

---

C:\Documents and Settings\Richard Katz>set
ALLUSERSPROFILE=C:\Documents and Settings\All Users
ANT_HOME=C:\cygwin\usr\local\apache-ant-1.6.2
APPDATA=C:\Documents and Settings\Richard Katz\Application Data
APR_ICONV_PATH=c:\programs\svn\iconv
CLASSPATH=.
CLIENTNAME=Console
CommonProgramFiles=C:\Program Files\Common Files
COMPUTERNAME=DC4FDBC1
ComSpec=C:\WINDOWS\system32\cmd.exe
CYGWIN=/cygdrive/c/cygwin
EMACSHOME=C:\programs\Emacs\
ERROR_CODE=1
FP_NO_HOST_CHECK=NO
G95_LIBRARY_PATH=C:\programs\g95\lib\gcc-lib\i686-pc-mingw32\4.1.2;C:\programs\g95\lib
HOME=C:\Documents and Settings\Richard Katz
HOMEDRIVE=C:
HOMEPATH=\Documents and Settings\Richard Katz
INCLUDE=C:\watcom-1.3\h;C:\watcom-1.3\h\nt;C:\watcom-1.3\maple\include;C:\watcom-1.3\h;C:\watcom-1.3\h\nt;C:\watcom-1.3\
maple\include
JAVA_HOME=C:\cygwin\opt\java\jdk1.6.0_01
JBOSS_HOME=C:\programs\jboss\jboss-4.2.2.GA
KMP_DUPLICATE_LIB_OK=TRUE
LOGONSERVER=\\DC4FDBC1
M2_HOME=C:\cygwin\usr\local\maven\apache-maven-2.1.0-M1
NUMBER_OF_PROCESSORS=2
OS=Windows_NT
OZEMACS=C:\programs\Emacs\bin\Emacs.exe
Path=C:\programs\Pure\;c:\ruby\bin;C:\programs\KomodoEdit4.2\;C:\programs\Emacs\\bin;C:\watcom-1.3\binnt;C:\watcom-1.3\b
inw;C:\watcom-1.3\binnt;C:\watcom-1.3\binw;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;c:\Python25;c:\Python
25\bin;c:\Python25\Scripts;C:\cygwin\usr\bin;c:\cygwin\bin;C:\programs\Mozart\bin;C:\programs\JGsoft\EditPadPro6;C:\cygw
in\opt\java\jdk1.6.0_01\bin;C:\programs\MySQL\MySQLServer5.0\bin;C:\cygwin\usr\local\maven\apache-maven-2.0.9\bin;c:\pro
grams\svn\bin;C:\cygwin\usr\local\maven\apache-maven-2.1.0-M1\bin;C:\Program Files\QuickTime\QTSystem\;C:\Program Files\
Haskell\bin;c:\programs\ghc\ghc683\bin;c:\programs\ssh;C:\programs\wscite;C:\programs\textpad;C:\cygwin\usr\local\apache
-ant-1.6.2\bin;c:\programs\PLT352;C:\programs\g95\bin
PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.RB;.RBW
PROCESSOR_ARCHITECTURE=x86
PROCESSOR_IDENTIFIER=x86 Family 6 Model 15 Stepping 6, GenuineIntel
PROCESSOR_LEVEL=6
PROCESSOR_REVISION=0f06
ProgramFiles=C:\Program Files
PROMPT=$P$G
PURELIB=C:\programs\Pure\lib
QTJAVA=C:\Program Files\Java\jre1.6.0_07\lib\ext\QTJava.zip
RUBYOPT=-rubygems
SESSIONNAME=Console
SonicCentral=C:\Program Files\Common Files\Sonic Shared\Sonic Central\
SUB_VOB_ROOT=c:/view/richarka_sub_view/ame-sub
SW=c:/dnld/sw
SystemDrive=C:
SystemRoot=C:\WINDOWS
TEMP=C:\DOCUME~1\RICHAR~1\LOCALS~1\Temp
TMP=C:\DOCUME~1\RICHAR~1\LOCALS~1\Temp
USER=richarka
USERDOMAIN=DC4FDBC1
USERNAME=Richard Katz
USERPROFILE=C:\Documents and Settings\Richard Katz
USER_HOME=C:\Documents and Settings\Richard Katz
VIEW=c:\view
WATCOM=C:\watcom-1.3
windir=C:\WINDOWS
XSB_BIN_DIRECTORY=C:\programs\xsbprolog\config\x86-pc-windows\bin\
XSB_DIR=C:\programs\xsbprolog

C:\Documents and Settings\Richard Katz>

---

Previous version of this issue (on 2.0.9) : 3809 was apparently closed on Dec 9, 2008.

Please let me know if there is any additional information I can provide.

**Comment by jira-importer** [comment]
**[Brett Porter](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brettporter)** commented

Richard, I suggest you utilise users@maven.apache.org where you can get better interactive help. You'll need to supply the -X output surrounding the line I quoted above.

**Comment by jira-importer** [comment]
**[Richard Katz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rich-katz)** commented

Please tell me what exactly I supposed to observe from the -U and -X options?
http://docs.codehaus.org/display/MAVENUSER/FAQs-1#FAQs-1-novalidversion)

I had already tried the -U option and have now tried the -X option as shown below.   This is really important.

Could you express in one or two sentences how to get around this problem?

Here is the result of the -X option:

---

C:\Documents and Settings\Richard Katz>mvn -X -U -e archetype:create -DgroupId=com.mycompany.app  -DartifactId=my-app
+ Error stacktraces are turned on.
  Maven version: 2.1.0-M1
  Java version: 1.6.0_01
  Default locale: en_US, platform encoding: Cp1252
  OS name: "windows xp" version: "5.1" arch: "x86" family: "windows"
  [DEBUG] Building Maven user-level plugin registry from: 'C:\Documents and Settings\Richard Katz\.m2\plugin-registry.xml'

[DEBUG] Building Maven global-level plugin registry from: 'C:\cygwin\usr\local\maven\apache-maven-2.1.0-M1\conf\plugin-r
egistry.xml'
[INFO] Scanning for projects...
[INFO] Searching repository for plugin with prefix: 'archetype'.
[DEBUG] Loading plugin prefixes from group: org.apache.maven.plugins
[INFO] org.apache.maven.plugins: checking for updates from central
[DEBUG] Using mirror: http://mirrors.ibiblio.org/pub/mirrors/maven2 (id: ibiblio.org)
[DEBUG] Checking for pre-existing User-Agent configuration.
[DEBUG] Adding User-Agent configuration.
[DEBUG] Connecting to repository: 'ibiblio.org' with url: 'http://mirrors.ibiblio.org/pub/mirrors/maven2'.
[WARNING] repository metadata for: 'org.apache.maven.plugins' could not be retrieved from repository: central due to an
error: Error transferring file
[DEBUG] Exception
org.apache.maven.wagon.TransferFailedException: Error transferring file
at org.apache.maven.wagon.providers.http.LightweightHttpWagon.fillInputData(LightweightHttpWagon.java:136)
at org.apache.maven.wagon.StreamWagon.getInputStream(StreamWagon.java:116)
at org.apache.maven.wagon.StreamWagon.getIfNewer(StreamWagon.java:88)
at org.apache.maven.wagon.StreamWagon.get(StreamWagon.java:61)
at org.apache.maven.artifact.manager.DefaultWagonManager.getRemoteFile(DefaultWagonManager.java:492)
at org.apache.maven.artifact.manager.DefaultWagonManager.getArtifactMetadata(DefaultWagonManager.java:389)
at org.apache.maven.artifact.repository.metadata.DefaultRepositoryMetadataManager.resolve(DefaultRepositoryMetad
ataManager.java:97)
at org.apache.maven.plugin.DefaultPluginMappingManager.loadPluginMappings(DefaultPluginMappingManager.java:103)
at org.apache.maven.plugin.DefaultPluginMappingManager.loadPluginMappings(DefaultPluginMappingManager.java:87)
at org.apache.maven.plugin.DefaultPluginMappingManager.getByPrefix(DefaultPluginMappingManager.java:61)
at org.apache.maven.plugin.DefaultPluginManager.getPluginDefinitionForPrefix(DefaultPluginManager.java:159)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.getMojoDescriptor(DefaultLifecycleExecutor.java:1718)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.segmentTaskListByAggregationNeeds(DefaultLifecycleExecuto
r.java:444)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.execute(DefaultLifecycleExecutor.java:176)
at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:336)
at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:129)
at org.apache.maven.cli.MavenCli.main(MavenCli.java:302)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
at java.lang.reflect.Method.invoke(Method.java:597)
at org.codehaus.classworlds.Launcher.launchEnhanced(Launcher.java:315)
at org.codehaus.classworlds.Launcher.launch(Launcher.java:255)
at org.codehaus.classworlds.Launcher.mainWithExitCode(Launcher.java:430)
at org.codehaus.classworlds.Launcher.main(Launcher.java:375)
Caused by: java.net.UnknownHostException: mirrors.ibiblio.org
at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:177)
at java.net.Socket.connect(Socket.java:519)
at java.net.Socket.connect(Socket.java:469)
at sun.net.NetworkClient.doConnect(NetworkClient.java:157)
at sun.net.www.http.HttpClient.openServer(HttpClient.java:388)
at sun.net.www.http.HttpClient.openServer(HttpClient.java:515)
at sun.net.www.http.HttpClient.\<init>(HttpClient.java:233)
at sun.net.www.http.HttpClient.New(HttpClient.java:306)
at sun.net.www.http.HttpClient.New(HttpClient.java:318)
at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:792)
at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:733)
at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:658)
at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:981)
at org.apache.maven.wagon.providers.http.LightweightHttpWagon.fillInputData(LightweightHttpWagon.java:115)
... 24 more
[INFO] Repository 'central' will be blacklisted
[DEBUG] Loading plugin prefixes from group: org.codehaus.mojo
[DEBUG] Skipping blacklisted repository central
[DEBUG] Skipping blacklisted repository central
[DEBUG] maven-archetype-plugin: using locally installed snapshot
[DEBUG] Artifact not found - using stub model: Unable to determine the latest version

org.apache.maven.plugins:maven-archetype-plugin:pom:LATEST

[DEBUG] Using defaults for missing POM org.apache.maven.plugins:maven-archetype-plugin:pom:LATEST
[DEBUG] maven-archetype-plugin: using locally installed snapshot
[DEBUG] Artifact not found - using stub model: Unable to determine the release version

org.apache.maven.plugins:maven-archetype-plugin:pom:RELEASE

[DEBUG] Using defaults for missing POM org.apache.maven.plugins:maven-archetype-plugin:pom:RELEASE
[INFO] ------------------------------------------------------------------------
[ERROR] BUILD ERROR
[INFO] ------------------------------------------------------------------------
[INFO] The plugin 'org.apache.maven.plugins:maven-archetype-plugin' does not exist or no valid version could be found
[INFO] ------------------------------------------------------------------------
[DEBUG] Trace
org.apache.maven.lifecycle.LifecycleExecutionException: The plugin 'org.apache.maven.plugins:maven-archetype-plugin' doe
s not exist or no valid version could be found
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.verifyPlugin(DefaultLifecycleExecutor.java:1553)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.getMojoDescriptor(DefaultLifecycleExecutor.java:1792)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.segmentTaskListByAggregationNeeds(DefaultLifecycleExecuto
r.java:444)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.execute(DefaultLifecycleExecutor.java:176)
at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:336)
at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:129)
at org.apache.maven.cli.MavenCli.main(MavenCli.java:302)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
at java.lang.reflect.Method.invoke(Method.java:597)
at org.codehaus.classworlds.Launcher.launchEnhanced(Launcher.java:315)
at org.codehaus.classworlds.Launcher.launch(Launcher.java:255)
at org.codehaus.classworlds.Launcher.mainWithExitCode(Launcher.java:430)
at org.codehaus.classworlds.Launcher.main(Launcher.java:375)
Caused by: org.apache.maven.plugin.version.PluginVersionNotFoundException: The plugin 'org.apache.maven.plugins:maven-ar
chetype-plugin' does not exist or no valid version could be found
at org.apache.maven.plugin.version.DefaultPluginVersionManager.resolvePluginVersion(DefaultPluginVersionManager.
java:229)
at org.apache.maven.plugin.version.DefaultPluginVersionManager.resolvePluginVersion(DefaultPluginVersionManager.
java:91)
at org.apache.maven.plugin.DefaultPluginManager.verifyPlugin(DefaultPluginManager.java:176)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.verifyPlugin(DefaultLifecycleExecutor.java:1524)
... 14 more
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3 seconds
[INFO] Finished at: Tue Feb 10 23:55:14 PST 2009
[INFO] Final Memory: 1M/4M
[INFO] ------------------------------------------------------------------------

C:\Documents and Settings\Richard Katz>m

**Comment by jira-importer** [comment]
**[Brett Porter](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brettporter)** commented

"Caused by: java.net.UnknownHostException: mirrors.ibiblio.org"

You're machine cannot reach mirrors.ibiblio.org

**Comment by jira-importer** [comment]
**[Richard Katz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rich-katz)** commented

Please tell me what exactly how I get around a Maven wagon Transfer exception or exactly what I am supposed to observe from either the -U and -X options?

http://docs.codehaus.org/display/MAVENUSER/FAQs-1#FAQs-1-novalidversion)

I had already tried the -U option and have now tried the -X option as shown below.

Here is the result of the -X option:

---

C:\Documents and Settings\Richard Katz>mvn -X -U -e archetype:create -DgroupId=com.mycompany.app  -DartifactId=my-app
+ Error stacktraces are turned on.
  Maven version: 2.1.0-M1
  Java version: 1.6.0_01
  Default locale: en_US, platform encoding: Cp1252
  OS name: "windows xp" version: "5.1" arch: "x86" family: "windows"
  [DEBUG] Building Maven user-level plugin registry from: 'C:\Documents and Settings\Richard Katz\.m2\plugin-registry.xml'

[DEBUG] Building Maven global-level plugin registry from: 'C:\cygwin\usr\local\maven\apache-maven-2.1.0-M1\conf\plugin-r
egistry.xml'
[INFO] Scanning for projects...
[INFO] Searching repository for plugin with prefix: 'archetype'.
[DEBUG] Loading plugin prefixes from group: org.apache.maven.plugins
[INFO] org.apache.maven.plugins: checking for updates from central
[DEBUG] Using mirror: http://mirrors.ibiblio.org/pub/mirrors/maven2 (id: ibiblio.org)
[DEBUG] Checking for pre-existing User-Agent configuration.
[DEBUG] Adding User-Agent configuration.
[DEBUG] Connecting to repository: 'ibiblio.org' with url: 'http://mirrors.ibiblio.org/pub/mirrors/maven2'.
[WARNING] repository metadata for: 'org.apache.maven.plugins' could not be retrieved from repository: central due to an
error: Error transferring file
[DEBUG] Exception
org.apache.maven.wagon.TransferFailedException: Error transferring file
at org.apache.maven.wagon.providers.http.LightweightHttpWagon.fillInputData(LightweightHttpWagon.java:136)
at org.apache.maven.wagon.StreamWagon.getInputStream(StreamWagon.java:116)
at org.apache.maven.wagon.StreamWagon.getIfNewer(StreamWagon.java:88)
at org.apache.maven.wagon.StreamWagon.get(StreamWagon.java:61)
at org.apache.maven.artifact.manager.DefaultWagonManager.getRemoteFile(DefaultWagonManager.java:492)
at org.apache.maven.artifact.manager.DefaultWagonManager.getArtifactMetadata(DefaultWagonManager.java:389)
at org.apache.maven.artifact.repository.metadata.DefaultRepositoryMetadataManager.resolve(DefaultRepositoryMetad
ataManager.java:97)
at org.apache.maven.plugin.DefaultPluginMappingManager.loadPluginMappings(DefaultPluginMappingManager.java:103)
at org.apache.maven.plugin.DefaultPluginMappingManager.loadPluginMappings(DefaultPluginMappingManager.java:87)
at org.apache.maven.plugin.DefaultPluginMappingManager.getByPrefix(DefaultPluginMappingManager.java:61)
at org.apache.maven.plugin.DefaultPluginManager.getPluginDefinitionForPrefix(DefaultPluginManager.java:159)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.getMojoDescriptor(DefaultLifecycleExecutor.java:1718)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.segmentTaskListByAggregationNeeds(DefaultLifecycleExecuto
r.java:444)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.execute(DefaultLifecycleExecutor.java:176)
at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:336)
at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:129)
at org.apache.maven.cli.MavenCli.main(MavenCli.java:302)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
at java.lang.reflect.Method.invoke(Method.java:597)
at org.codehaus.classworlds.Launcher.launchEnhanced(Launcher.java:315)
at org.codehaus.classworlds.Launcher.launch(Launcher.java:255)
at org.codehaus.classworlds.Launcher.mainWithExitCode(Launcher.java:430)
at org.codehaus.classworlds.Launcher.main(Launcher.java:375)
Caused by: java.net.UnknownHostException: mirrors.ibiblio.org
at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:177)
at java.net.Socket.connect(Socket.java:519)
at java.net.Socket.connect(Socket.java:469)
at sun.net.NetworkClient.doConnect(NetworkClient.java:157)
at sun.net.www.http.HttpClient.openServer(HttpClient.java:388)
at sun.net.www.http.HttpClient.openServer(HttpClient.java:515)
at sun.net.www.http.HttpClient.\<init>(HttpClient.java:233)
at sun.net.www.http.HttpClient.New(HttpClient.java:306)
at sun.net.www.http.HttpClient.New(HttpClient.java:318)
at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:792)
at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:733)
at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:658)
at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:981)
at org.apache.maven.wagon.providers.http.LightweightHttpWagon.fillInputData(LightweightHttpWagon.java:115)
... 24 more
[INFO] Repository 'central' will be blacklisted
[DEBUG] Loading plugin prefixes from group: org.codehaus.mojo
[DEBUG] Skipping blacklisted repository central
[DEBUG] Skipping blacklisted repository central
[DEBUG] maven-archetype-plugin: using locally installed snapshot
[DEBUG] Artifact not found - using stub model: Unable to determine the latest version

org.apache.maven.plugins:maven-archetype-plugin:pom:LATEST

[DEBUG] Using defaults for missing POM org.apache.maven.plugins:maven-archetype-plugin:pom:LATEST
[DEBUG] maven-archetype-plugin: using locally installed snapshot
[DEBUG] Artifact not found - using stub model: Unable to determine the release version

org.apache.maven.plugins:maven-archetype-plugin:pom:RELEASE

[DEBUG] Using defaults for missing POM org.apache.maven.plugins:maven-archetype-plugin:pom:RELEASE
[INFO] ------------------------------------------------------------------------
[ERROR] BUILD ERROR
[INFO] ------------------------------------------------------------------------
[INFO] The plugin 'org.apache.maven.plugins:maven-archetype-plugin' does not exist or no valid version could be found
[INFO] ------------------------------------------------------------------------
[DEBUG] Trace
org.apache.maven.lifecycle.LifecycleExecutionException: The plugin 'org.apache.maven.plugins:maven-archetype-plugin' doe
s not exist or no valid version could be found
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.verifyPlugin(DefaultLifecycleExecutor.java:1553)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.getMojoDescriptor(DefaultLifecycleExecutor.java:1792)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.segmentTaskListByAggregationNeeds(DefaultLifecycleExecuto
r.java:444)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.execute(DefaultLifecycleExecutor.java:176)
at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:336)
at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:129)
at org.apache.maven.cli.MavenCli.main(MavenCli.java:302)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
at java.lang.reflect.Method.invoke(Method.java:597)
at org.codehaus.classworlds.Launcher.launchEnhanced(Launcher.java:315)
at org.codehaus.classworlds.Launcher.launch(Launcher.java:255)
at org.codehaus.classworlds.Launcher.mainWithExitCode(Launcher.java:430)
at org.codehaus.classworlds.Launcher.main(Launcher.java:375)
Caused by: org.apache.maven.plugin.version.PluginVersionNotFoundException: The plugin 'org.apache.maven.plugins:maven-ar
chetype-plugin' does not exist or no valid version could be found
at org.apache.maven.plugin.version.DefaultPluginVersionManager.resolvePluginVersion(DefaultPluginVersionManager.
java:229)
at org.apache.maven.plugin.version.DefaultPluginVersionManager.resolvePluginVersion(DefaultPluginVersionManager.
java:91)
at org.apache.maven.plugin.DefaultPluginManager.verifyPlugin(DefaultPluginManager.java:176)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.verifyPlugin(DefaultLifecycleExecutor.java:1524)
... 14 more
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3 seconds
[INFO] Finished at: Tue Feb 10 23:55:14 PST 2009
[INFO] Final Memory: 1M/4M
[INFO] ------------------------------------------------------------------------

C:\Documents and Settings\Richard Katz>m

This absolutely is a problem and the exception produced is a Maven exception, not a Java exception.

1. What libraries are you using to try to read this file that produce "org.apache.maven.wagon.TransferFailedException?"

2. Why does it say "Repository 'central' will be blacklisted?"  What does this message indicate that a user should do?

There is no clear way to cause Maven 2 to correctly read from the repository that I know of and yet I read files over HTTP 24 x 7 here.  And this is getting a Maven exception - not a Java I/O exception.

This is really important.    Could you express in one or two sentences how to get around this problem or what to do about a Maven wagon exception?

**Comment by jira-importer** [comment]
**[Richard Katz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rich-katz)** commented

Is this an e-mail address I am supposed to send to?

Richard, I suggest you utilise users@maven.apache.org where you can get better interactive help. You'll need to supply the -X output surrounding the line I quoted above.

**Comment by jira-importer** [comment]
**[Richard Katz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rich-katz)** commented

"Caused by: java.net.UnknownHostException: mirrors.ibiblio.org"

You're machine cannot reach mirrors.ibiblio.org

Yes it can!!

http://mirrors.ibiblio.org/

Index of /

Icon  Name                    Last modified      Size  Description[DIR] pub/                    23-May-2007 14:59    -

Apache Server at mirrors.ibiblio.org Port 80

**Comment by jira-importer** [comment]
**[Brett Porter](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brettporter)** commented

does your browser use a proxy setting that you have not configured for maven, per the link shared earlier?

**Comment by jira-importer** [comment]
**[Richard Katz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rich-katz)** commented

Browsing to the directory named in settings.xml - works!

**Comment by jira-importer** [comment]
**[Richard Katz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rich-katz)** commented

OK.

I found it.  I got it to work.

I just need one thing.  What setting should be changed in a personal firewall (not a proxy but a personal firewall) such as Norton  Internet Security to get Maven 2 to work?

See:
http://mail-archives.apache.org/mod_mbox/maven-users/200705.mbox/%3C464729C6.2030303@alltel.net%3E

There are about 5 copies of this floating around.  Was there a resolution (such as "turn on port xxx" for Java/Maven/etc...).

If not, that's OK too.

Thanks for your fast responses.

Sincerely,

Rich
